src/components/WeatherWidget.vue

```diff
@@ -1,41 +1,101 @@
 <template>
   <div class="weather-widget">
-    <WeatherWidgetPlace
-      v-bind="weatherData.location"
-      :date="selectedDayData?.date ?? 'Today'"
+    <WeatherWidgetMessage 
+      v-if="globalIsLoading"
+      :message="'Loading...'" 
     />
-    <WeatherWidgetCurrentData v-bind="selectedDayData" />
-    <WeatherWidgetForecast
-      :forecastData="weatherData.forecast"
-      :selectedDay
-      @handleDayChange="handleDayChange"
+    <WeatherWidgetMessage
```

Comment:
I think we can have 1 WeatherWidgetMessage displaying different mesage depending on state, you can create a computed for it